spec/features/work_packages/table/queries/query_menu_spec.rb

```diff
@@ -29,26 +29,35 @@
 require 'spec_helper'
 
 describe 'Query menu item', js: true do
-  let(:user) { create :admin }
   let(:project) { create :project }
   let(:wp_table) { ::Pages::WorkPackagesTable.new(project) }
   let(:filters) { ::Components::WorkPackages::Filters.new }
   let(:query_title) { ::Components::WorkPackages::QueryTitle.new }
 
-  before do
-    login_as(user)
-  end
+  current_user { create :admin }
 
-  context 'visiting the global work packages page' do
+  context 'when visiting the global work packages page' do
     let(:wp_table) { ::Pages::WorkPackagesTable.new }
-    it 'should show the query menu (Regression #30082)' do
+    let(:project) { nil }
+
+    let!(:global_query) { create(:query, project: nil) }
+    let!(:project_query) { create(:query, project: create(:project)) }
+
+    it 'shows the query menu with queries stored for the global page' do
```

Comment:
This test is currently failing on the CI.

As far as I understand, there are two issues:
- only the queries are created in the test. When the work package page is displayed, endpoint `/api/v3/views?[{"type":{"operator":"=","values":["Views::WorkPackagesTable"]}},{"project":{"operator":"!*","values":[]}}]` gets called but there are no views attached to the query, so it return an empty array and the test fails (global query not visible). Using `create(:query_with_view_work_packages_table)` instead of `create(:query)` fixes it
- the created query are not owned by the current user, so `Query.visible(User.current)` will return `[]` (as fixed by Oliver in #10472). Adding `user: current_user to the factory fixes it.

I pushed a fix for that, but please review if that's correct.

Reply:
Hi @cbliard , you are correct. I did work on a more elaborate spec this morning without checking your fix with a few more cases tested. So I allowed myself the audacity to just force push over your spec fix.

Reply:
Yep, it's better, although for the `project_view` the query does not belong to the current_user, and the current_user is not a member of the project so the view will never be displayed to them. So the test does not really test that project queries/views are not displayed on global work package page.